core/kazoo_bindings/src/kazoo_bindings.erl

```diff
@@ -710,7 +713,9 @@ map_processor_fold(#kz_binding{binding_parts=BParts
                   ,_Routing
                   ,RoutingParts
                   ) ->
-    case matches(BParts, RoutingParts) of
+    case matches(BParts, RoutingParts)
+        orelse matches(RoutingParts, BParts)
```

Comment:
kazoo_bindings worked in the premise that the routing key passed for processing is concrete, however some operations may actually want to run all possible bindings based on wildcards. `tasks.help.*` is one of them and `flush` within kazoo_bindings is another (cleaning "tasks.#" on startup). the `orelse` added makes this happen. 

@jamesaimonetti since matches/1 and candidates/1 help us to debug these, you can try
kazoo_bindings:matches(<<"tasks.help.*">>) with and without the `orelse`